go/libraries/doltcore/diff/table_deltas.go

```diff
@@ -257,6 +259,32 @@ func GetTableDeltas(ctx context.Context, fromRoot, toRoot *doltdb.RootValue) (de
 		})
 	}
 
+	sort.Slice(deltas, func(i, j int) bool {
```

Comment:
It seems somewhat fragile to have the staging logic in `MoveTablesBetweenRoots` depend on the sorting order here. See comment below